include/wil/registry_helpers.h

```diff
@@ -1218,6 +1218,531 @@ namespace wil
+            // if more string values are desired, all 5 functions must be implement for that type:
+            // clear_name, copy_name, compare_name, resize_name, address_of_name
+            inline void clear_name(::wil::unique_process_heap_string& name, size_t length)
+            {
+                if (length > 0)
+                {
+                    memset(name.get(), 0, length * sizeof(wchar_t));
+                }
+            }
+            inline ::wil::unique_process_heap_string copy_name(const ::wil::unique_process_heap_string& str, size_t capacity)
+            {
+                if (!str)
+                {
+                    return {};
+
+                }
+                return ::wil::make_process_heap_string(str.get(), capacity);
+            }
+            inline bool compare_name(const ::wil::unique_process_heap_string& name, PCWSTR comparand) WI_NOEXCEPT
+            {
+                if (!name)
+                {
+                    return false;
+                }
+                return 0 == wcscmp(name.get(), comparand);
+            }
+            inline size_t resize_name(::wil::unique_process_heap_string& name, size_t current_length, size_t new_length)
+            {
+                if (new_length > current_length)
+                {
+                    auto newString{ ::wil::make_unique_string_nothrow<::wil::unique_process_heap_string>(name.get(), new_length) };
+                    name.swap(newString);
+                    return new_length;
+                }
+
+                // continue to use the existing buffer since the requested length is less than or equals to the current length
+                clear_name(name, current_length);
+                return current_length;
+            }
+            inline PWSTR address_of_name(::wil::unique_process_heap_string& name) WI_NOEXCEPT
+            {
+                return name.get();
+            }
+
+#if defined(_STRING_)
+            inline void clear_name(::std::wstring& name, size_t)
+            {
+                name.assign(name.size(), L'\0');
+            }
+            inline ::std::wstring copy_name(const ::std::wstring& str, size_t)
+            {
+                return str;
+            }
+            inline bool compare_name(const ::std::wstring& name, PCWSTR comparand) WI_NOEXCEPT
+            {
+                // not using operat== as that creates a temp std::wstring, which could fail/throw
+                return 0 == wcscmp(name.c_str(), comparand);
+            }
+            inline size_t resize_name(::std::wstring& name, size_t, size_t new_length)
+            {
+                name.resize(new_length);
+                clear_name(name, name.size());
+                return name.size();
+            }
+            inline PWSTR address_of_name(::std::wstring& name) WI_NOEXCEPT
+            {
+                // cannot return a const PCWSTR -- the registry APIs require a PWSTR
+                return const_cast<PWSTR>(name.data());

```

Comment:
`std::basic_string::data` already returns a non-`const` pointer. The `const_cast` is unnecessary